### PR TITLE
Create functionality for send selling request 

### LINKED ISSUE #76: Create adding a car functionality
This feature should allow the customer to add selling car request.
- In this task:
   - build controller function.
   - build service function that add to database.
   - add the endpoint to the car's routes.

### PR DESCRIPTION
**Pull request by tariqabusada**
- [x] rearrange server files
- [x] create 'addCar' function in controllers
- [x] create 'addCarService' function in services
- [x] create fetch in client side to send the request to server
- [x] create route to run the middlewares

### CODE REVIEW

File: client/src/routes/index.tsx [hunk 1]
```diff
@@ -3,7 +3,6 @@ import SignUp from '../pages/SignUp';
 import Login from '../pages/Login';
 import AdminLogin from '../pages/AdminLogin';
 import Profile from '../pages/Profile';
-// import Cars from '../pages/Cars';
```
Comment (abdou059) [inline client/src/routes/index.tsx:6]: remove comment

File: server/src/validation/updateCarSchema.ts [hunk 2]
```diff
@@ -14,6 +14,8 @@ const updateCarSchema = yup.object().shape({
   description: yup.string().notRequired().label('description'),
   fuel: yup.string().notRequired().label('fuel').oneOf(['petrol', 'diesel']),
   state: yup.string().notRequired().label('state').oneOf(['under-check', 'pending', 'on-market']),
+  location: yup.string().notRequired().label('location'),
+  customerId: yup.number().notRequired().label('customerId'),
```
Comment (LinaYahya) [inline server/src/validation/updateCarSchema.ts]: not sure why you added `customerId` here?

File: server/src/services/cars/addCar.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Car } from '../../db/models';
+
+const addCarService = async (body, customerId) => {
+  const carRequest = await Car.bulkCreate(
```
Comment (LinaYahya) [inline server/src/services/cars/addCar.ts]: bulk create used to insert more than a row to db, But we're not doing that right now,
check `Model.create`

File: server/src/routes/cars.ts [hunk 4]
```diff
@@ -12,4 +17,6 @@ carsRouter.get('/', ExpressWrapper(getCarsById));
 carsRouter.get('/dashboard', authMiddleware('admin'), ExpressWrapper(getCarsDetails));
 carsRouter.delete('/:id', authMiddleware('admin'), ExpressWrapper(deleteCarsById));
 carsRouter.put('/:id', authMiddleware('admin'), ExpressWrapper(updateCar));
+// carsRouter.get('/profile/sell', authMiddleware('user'), ExpressWrapper(addCar));
```
Comment (LinaYahya) [inline server/src/routes/cars.ts]: not sure where's the route?

File: server/src/controllers/cars/addCar.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { Request } from 'express';
+import { addCarService } from '../../services';
+import { updateCarSchema } from '../../validation';
+
+const addCar = async (req: Request) => {
+  const customerId = req.cookies.id;
```
Comment (LinaYahya) [inline server/src/controllers/cars/addCar.ts]:
```suggestion
  const customerId = req.cookies.id; 
```
DO we have `id` at `cookies`?

File: server/src/controllers/cars/addCar.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  const { body } = req;
+  await updateCarSchema.validate(body, customerId);
```
Comment (LinaYahya) [inline server/src/controllers/cars/addCar.ts]:
```suggestion
  await updateCarSchema.validate(body, customerId);
```
Does validate accept two params?

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: client/src/components/sendRequsetModel/Form.tsx [hunk 7]
```diff
@@ -34,9 +35,16 @@ function SellCarModal() {
       const newValue = Math.floor(convertToKM(values.milage, values.type));
       // eslint-disable-next-line no-param-reassign
       values.milage = newValue;
-      console.log(values);
+      // console.log(values);
```
Comment (husamkamal) [inline client/src/components/sendRequsetModel/Form.tsx]: remove

File: server/src/controllers/cars/addCar.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { Request } from 'express';
+import { addCarService } from '../../services';
+import { addCarSchema } from '../../validation';
+
+const addCar = async (req: Request, res) => {
+  // const { token } = req.cookies;
```
Comment (husamkamal) [inline server/src/controllers/cars/addCar.ts]: remove

File: server/src/services/cars/addCar.ts [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import { Car } from '../../db/models';
+
+const addCarService = async (body, id) => {
```
Comment (LinaYahya) [inline server/src/services/cars/addCar.ts]:
```suggestion
const addCarService = async (data) => { // {...body, customerId}
```

File: server/src/services/cars/addCar.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+  const carRequest = await Car.create(
```
Comment (LinaYahya) [inline server/src/services/cars/addCar.ts]:
```suggestion
  const car = await Car.create(
```
It's only  a car

File: server/src/routes/cars.ts [hunk 11]
```diff
@@ -12,4 +17,6 @@ carsRouter.get('/', ExpressWrapper(getCarsById));
 carsRouter.get('/dashboard', authMiddleware('admin'), ExpressWrapper(getCarsDetails));
 carsRouter.delete('/:id', authMiddleware('admin'), ExpressWrapper(deleteCarsById));
 carsRouter.put('/:id', authMiddleware('admin'), ExpressWrapper(updateCar));
+carsRouter.post('/sell-car', authMiddleware('user'), ExpressWrapper(addCar));
```
Comment (LinaYahya) [inline server/src/routes/cars.ts]:
```suggestion
carsRouter.post('/', authMiddleware('user'), ExpressWrapper(addCar));
```
it's not a good practice to have a resource name that include the method,
request method `post` will definitely means it's add

File: server/src/controllers/cars/addCar.ts [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+  const { body } = req;
+  const { userId } = res.locals.user;
+  await addCarSchema.validate(body);
+  const result = await addCarService(body, userId);
+  return { message: 'successfully', status: 200, data: result };
```
Comment (LinaYahya) [inline server/src/controllers/cars/addCar.ts]:
```suggestion
  return { message: 'successfully', status: 201, data: result };
```

File: client/src/components/sendRequsetModel/Form.tsx [hunk 13]
```diff
@@ -34,9 +35,16 @@ function SellCarModal() {
...
+      httpInstance.post(
+        '/cars/sell-car',
+        values,
+      )
+        .catch((err: Error) => {
+          // eslint-disable-next-line no-console
+          console.log(err);
+        });
```
Comment (LinaYahya) [inline client/src/components/sendRequsetModel/Form.tsx]: here we need to handle `loading`, and `success mesaage` and `error message`

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by husamkamal — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by abdou059 — APPROVED** [review]

**Review by husamkamal — APPROVED** [review]

**Review by mai-mohammed — APPROVED** [review]

**Pull request merged**